packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -254,79 +268,83 @@ const helperBehavior = {
+        .catch(e => console.log('Liquidator failed', e))
+        .finally(() => {
+          liquidationInProgress = false;
+        });
+    }
 
-    // INTERLOCK NOTE:
-    // The presence of an `outstandingQuote` is used to ensure there's only
-    // one thread of control waiting for a quote from the price authority.
-    // All later callers will just update the threshold for the `outstandingQuote`.
-    // The first invocation will skip the next block and continue to below where
-    // the `outstandingQuote` is assigned. So, if there's an outstanding quote,
-    // reset the level. If there's no current quote (because this is the first
-    // vault, or because a quote just resolved), then make a new request to the
-    // priceAuthority, and when it resolves, liquidate the first vault.
-    if (state.outstandingQuote) {
+    // There is already an activity processing liquidations. It may be
+    // waiting for the oracle price to cross a threshold.
+    // Update the current in-progress quote if there is one. Otherwise,
+    // the new threshold will be picked up by the next quote request.
+    if (outstandingQuote) {
+      const govParams = state.factoryPowers.getGovernedParams();
+      const liquidationMargin = govParams.getLiquidationMargin();
       // Safe to call extraneously (lightweight and idempotent)
-      E(state.outstandingQuote).updateLevel(
+      E(outstandingQuote).updateLevel(
         highestDebtRatio.denominator, // collateral
-        triggerPoint,
-      );
-      trace(
-        'updating level for outstandingQuote',
-        triggerPoint,
-        highestDebtRatio.denominator,
+        liquidationThreshold(highestDebtRatio, liquidationMargin),
       );
-      return;
+      trace('update quote', highestDebtRatio);
     }
+  },
 
-    if (state.liquidationInProgress) {
-      return;
+  processLiquidations: async ({ state, facets }) => {
+    const { prioritizedVaults, priceAuthority } = state;
+    const govParams = state.factoryPowers.getGovernedParams();
+
+    async function* eventualLiquidations() {
+      while (true) {
+        const highestDebtRatio = prioritizedVaults.highestRatio();
+        if (!highestDebtRatio) {
+          return;
+        }
+        const liquidationMargin = govParams.getLiquidationMargin();
+
+        // ask to be alerted when the price level falls enough that the vault
+        // with the highest debt to collateral ratio will no longer be valued at the
+        // liquidationMargin above its debt.
+        outstandingQuote = E(priceAuthority).mutableQuoteWhenLT(
+          highestDebtRatio.denominator, // collateral
+          liquidationThreshold(highestDebtRatio, liquidationMargin),
+        );
+        trace('posted quote request', highestDebtRatio);
+
+        // The rest of this method will not happen until after a quote is received.
+        // This may not happen until much later, when the market changes.
+        // eslint-disable-next-line no-await-in-loop
```

Comment:
thoughts on disabling this lint rule?

Reply:
Nope. The `await` here carefully doesn't violate our driving rule, but typical usages would. Given that at least one of the `reschedulePriceCheck` bugs was a bogus `await` violation, it's important to internalize the underlying rule and enforce it pretty rigorously.